### PR TITLE
one-line fix for a minor bug about encoding 

### PR DESCRIPTION
**Pull request by ch-yx**
i think it is necessary to make encoding and decoding use the same codec

it is ok that both of them use local encoding.
it is ok that both of them use the same specific encoding.
it is not ok that only one of them use local encoding.

### CODE REVIEW

**Comment by Ghoulboy78** [comment]
IMO it's better to use some common encoding for both, like UTF-8, so that people can send messages in base64 and then decode them and have them make sense. I'm also a little embarrassed that I didn't notice this when I wrote this code.

**Comment by Ghoulboy78** [comment]
I would also recommend changing a few lines down in the json section. I have noticed that we use a deprecated method to parse Json, so I would recommend to replace line 344 which currently is:
`return new ScarpetJsonDeserializer().deserialize(new JsonParser().parse(v.getString()), null, null);`
With:
`return new ScarpetJsonDeserializer().deserialize(JsonParser.parseString(v.getString()), null, null);`

File: src/main/java/carpet/script/language/DataStructures.java [hunk 1]
```diff
@@ -341,7 +341,7 @@ else if (lv.size() == 2)
         expression.addUnaryFunction("encode_json", v -> StringValue.of(v.toJson().toString()));
         expression.addUnaryFunction("decode_json", v -> {
             try {
-                return new ScarpetJsonDeserializer().deserialize(new JsonParser().parse(v.getString()), null, null);
+                return new ScarpetJsonDeserializer().deserialize(JsonParser.parseString(v.getString()), null, null);
```
Comment (altrisi) [inline src/main/java/carpet/script/language/DataStructures.java:344]: This is wrong, the deserializer shouldn't be called by just nulling out parameters that may be required later. Not only that, this warning is already fixed in another PR.

Comment (ch-yx) [reply src/main/java/carpet/script/language/DataStructures.java]: ok, i revert the latest commit.

**Review by altrisi — APPROVED** [review]

**Comment by ch-yx** [comment]
thank you a lot for your approve!
now, does it only need gnembon's approve to be merged?

**Comment by gnembon** [comment]
b64 is not using anything outside of ascii so encoding don't matter, but hey. If it doesn't matter it can be utf_8

**Comment by ch-yx** [comment]
> b64 is not using anything outside of ascii so encoding don't matter, but hey. If it doesn't matter it can be utf_8

b64 itself ofc is ascii only. but the string the function received/yield could be out of ascii.

before that is merged, decode_b64(encode_b64('我')) is not ‘我’ and the result is depends on the os `s default encoding.

surely that is a bug. and im glad that is fixed.

**Pull request merged**